include/bluetooth/bluetooth.h

```diff
@@ -513,6 +516,14 @@ enum {
 
 	/** Disable advertising on channel index 39. */
 	BT_LE_ADV_OPT_DISABLE_CHAN_39 = BIT(17),
+
+	/** Put GAP device name into advert data
+	 *
+	 *  Requires @ref BT_LE_ADV_OPT_USE_NAME to be set. Will place the
+	 *  GAP device name into the advertising data rather than the scan
+	 *  response data.
+	 */
+	BT_LE_ADV_OPT_FORCE_NAME_IN_AD = BIT(18),
```

Comment:
If what @jhedberg says that this is not the recommended way of doing it, I think it's worth mentioning here.

Reply:
I wouldn't say it's not the recommended way of doing it, it's very much down to the application. If you've got a single service and want to advertise that with the device name and it all fits into the advertising data, it should go there, it doesn't actually make any sense in that scenario to put it in the scan response nor does it make any sense to have any scan response data at all as all that would do is waste battery life if it's a sensor. However, if you're advertising multiple 128-bit UUID services then that will use a lot of advert space and it would be more important to see what services were being advertised rather than the device name.